sdks/python/container/boot.go

```diff
@@ -145,7 +145,21 @@ func main() {
 	// Guard from concurrent artifact retrieval and installation,
 	// when called by child processes in a worker pool.
 
+	workerPoolId := os.Getenv(workerPoolIdEnv)
+	var venvDir string
+	if workerPoolId != "" {
+		venvDir = filepath.Join(*semiPersistDir, "beam-venv", "beam-pool-" + workerPoolId)
+	} else {
+		venvDir = filepath.Join(*semiPersistDir, "beam-venv", "beam-worker-" + *id)
+	}
```

Comment:
As mentioned above, this is super strange. I would much prefer to have a job ID here.

Reply:
Alternatively, we could always bind this to the worker ID and remove the whole `multiProcessExactlyOnce()` logic, but I don't know if that's important for anything.

Reply:
Judging by this comment, nixing the exactly-once stuff wouldn't be an issue as long as each worker installs to a separate venv: https://github.com/apache/beam/pull/9371#discussion_r316351902

Reply:
Whether a job ID or is or is not available may be runner-specific.